cmd/omni/cmd/run.go

```diff
@@ -8,46 +8,64 @@ package cmd
 import (
 	"context"
 	"fmt"
-	"strings"
 
+	"dario.cat/mergo"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
 	"github.com/go-logr/zapr"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/siderolabs/talos/pkg/machinery/config/generate"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
-	"gopkg.in/yaml.v3"
 	"k8s.io/klog/v2"
 
 	"github.com/siderolabs/omni/client/pkg/compression"
 	"github.com/siderolabs/omni/client/pkg/constants"
+	authres "github.com/siderolabs/omni/client/pkg/omni/resources/auth"
+	omnires "github.com/siderolabs/omni/client/pkg/omni/resources/omni"
 	"github.com/siderolabs/omni/client/pkg/panichandler"
+	"github.com/siderolabs/omni/internal/backend"
+	"github.com/siderolabs/omni/internal/backend/discovery"
+	"github.com/siderolabs/omni/internal/backend/dns"
+	"github.com/siderolabs/omni/internal/backend/imagefactory"
 	"github.com/siderolabs/omni/internal/backend/logging"
+	"github.com/siderolabs/omni/internal/backend/resourcelogger"
 	"github.com/siderolabs/omni/internal/backend/runtime/omni"
+	"github.com/siderolabs/omni/internal/backend/runtime/omni/virtual"
+	"github.com/siderolabs/omni/internal/backend/runtime/talos"
+	"github.com/siderolabs/omni/internal/backend/workloadproxy"
+	"github.com/siderolabs/omni/internal/pkg/auth"
 	"github.com/siderolabs/omni/internal/pkg/auth/actor"
+	"github.com/siderolabs/omni/internal/pkg/auth/user"
 	"github.com/siderolabs/omni/internal/pkg/config"
+	"github.com/siderolabs/omni/internal/pkg/ctxstore"
+	"github.com/siderolabs/omni/internal/pkg/features"
+	"github.com/siderolabs/omni/internal/pkg/siderolink"
 	"github.com/siderolabs/omni/internal/version"
 )
 
 // RunService starts the main Omni server.
-func RunService(ctx context.Context, logger *zap.Logger, params *config.Params) error {
+func RunService(ctx context.Context, logger *zap.Logger, paramsList ...*config.Params) error {
 	cfg := config.InitDefault()
 
-	raw, err := yaml.Marshal(params)
-	if err != nil {
-		return err
+	for _, params := range paramsList {
+		if err := mergo.Merge(cfg, params, mergo.WithOverride); err != nil {
```

Comment:
we do have `merge` in Talos `machinery` package, you might want to use this instead of pulling a dependency

Reply:
are you talking about https://github.com/siderolabs/talos/blob/81ca27949427c546f43b0409b56f733becabc2f6/pkg/machinery/config/configpatcher/strategic.go#L31 ?
This one is too specific for the configs